### ISSUE TITLE
iPad Testing Issues

### ISSUE THREAD

**Issue by kyleblythe** [sakaiproject/sakai#1486]
Cannot access the “Add Comment” modal window after opening the cell drop-down: https://www.evernote.com/l/ANxrJkElDXNKMbN0PsEUlL7kzDglr3HaeEc

Cannot scroll inside Grade Summary screen: https://www.evernote.com/l/ANygCDlIBVpLz4DaDvShFxy_sqMFgA32U3w

Generally, lots of lag when scrolling through the columns

**Comment by payten** [comment]
Hi @kyleblythe, I've had a look at these and unfortunately for the 2 latter issues there's no quick fix:

>  Cannot access the “Add Comment” modal window after opening the cell drop-down

I think this is fixed now.  We recently applied a fix to resolve this for Firefox and I think that does the job here.

> Cannot scroll inside Grade Summary screen

This is a tricky one!  With the move to a responsive design, we can't really distinguish ipad or touch devices from standard browsers anymore.  The only indicators we can use to modify the styles are height and width of the viewport.  So unfortunately there's no easy fix here.  We could force the scrollbars don't display for smaller viewports, however it's not a perfect solution as a landscape oriented ipad is 1024px wide which is the same as the browser on my smaller MacBook. 

We added those scrollbars so the actions at the bottom of the dialog are always visible, so we can remove them completely and perhaps move the actions to the top of the page as well as the bottom so they are still accessible?

> Generally, lots of lag when scrolling through the columns

Currently, we reposition the headers for every pixel the user scrolls, which causes quite a few renders and subsequently feels sluggish in some browsers.

I think we can fix this by only rendering the fixed headers and columns after the user has finished scrolling.  One implementation would be we hide the fixed headers while the user is scrolling scrolling and then render them again in their new position a few tenths of second after the user finishes scrolling.  As mentioned above, we can't target iPads, so it would change the feel for both desktop and touch-screen users.

I can mock this up for you to review?

**Comment by kyleblythe** [comment]
Hi @payten,

Hmm, this is a tricky one. Thanks for thinking through these!

That's a shame re: the action buttons in the "Grade Summary" screen - I like that they are always visible at the bottom of the window, and think that the screen looks cleaner without the duplication of elements. The scrolling is working really well (at least, on a non-mobile device).

And re: the headers - I don't know if I'm comfortable with the idea of those fixed headers disappearing as the user scrolls. This works beautifully on non-mobile devices at the moment, and I can only imagine the frustration instructors would experience as they try to scroll through the interface to find a given score, only to have the student names/item header disappear while they scroll. What do you think, @jeffpasch?

**Comment by payten** [comment]
Oo, so it looks like the scroll issue might be a Bootstrap bug: (github.com/twbs/bootstrap/issues/14839).  They've marked it as a "Won't Fix", but there are quite a few workarounds in the comments.  I'll investigate.

**Comment by payten** [comment]
Hi @kyleblythe, I've submitted a PR (#1600) that adds the accelerated scroll to all scrollable areas in the gradebook grades page - namely this property `-webkit-overflow-scrolling: touch` (https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-overflow-scrolling).  With this CSS property, scrolling feels a lot more natural, while being a little bit choppy it's still usable.

I've also fixed an issue with the Previous/Next student actions in the summary modal whereby they couldn't be clicked on a smaller viewport.

Will poke when it's merged!
